File: kalexa-model/src/main/java/com/hp/kalexa/model/services/ListService.java
/*
 * Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
 * this file except in compliance with the License. A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.hp.kalexa.model.services;


import com.hp.kalexa.model.request.list.*;

public interface ListService {

    AlexaListsMetadata getListsMetadata(String token) throws ServiceException;

    AlexaList getList(String listId, ListState status, String token) throws ServiceException;

    AlexaListMetadata createList(CreateListRequest request, String token) throws ServiceException;

    AlexaListMetadata updateList(String listId, UpdateListRequest request, String token) throws ServiceException;

    void deleteList(String listId, String token) throws ServiceException;

    AlexaListItem getListItem(String listId, String itemId, String token) throws ServiceException;

    AlexaListItem createListItem(String listId, CreateListItemRequest request, String token) throws ServiceException;

    AlexaListItem updateListItem(String listId, String itemId, UpdateListItemRequest request, String token) throws ServiceException;

    void deleteListItem(String listId, String itemId, String token) throws ServiceException;

}

File: kalexa-model/src/main/kotlin/com/hp/kalexa/model/services/list/ListService.kt
/*
 * Copyright 2018 HP Development Company, L.P.
 * SPDX-License-Identifier: MIT
 */

package com.hp.kalexa.model.services.list

import com.hp.kalexa.model.request.list.AlexaList
import com.hp.kalexa.model.request.list.AlexaListItem
import com.hp.kalexa.model.request.list.AlexaListMetadata
import com.hp.kalexa.model.request.list.AlexaListsMetadata
import com.hp.kalexa.model.request.list.CreateListItemRequest
import com.hp.kalexa.model.request.list.CreateListRequest
import com.hp.kalexa.model.request.list.ListState
import com.hp.kalexa.model.request.list.UpdateListItemRequest
import com.hp.kalexa.model.request.list.UpdateListRequest
import com.hp.kalexa.model.services.ServiceException

interface ListService {

    @Throws(ServiceException::class)
    fun getListsMetadata(): AlexaListsMetadata

    @Throws(ServiceException::class)
    fun getList(listId: String, status: ListState): AlexaList

    @Throws(ServiceException::class)
    fun createList(request: CreateListRequest): AlexaListMetadata

    @Throws(ServiceException::class)
    fun updateList(listId: String, request: UpdateListRequest): AlexaListMetadata

    @Throws(ServiceException::class)
    fun deleteList(listId: String)

    @Throws(ServiceException::class)
    fun getListItem(listId: String, itemId: String): AlexaListItem

    @Throws(ServiceException::class)
    fun createListItem(listId: String, request: CreateListItemRequest): AlexaListItem

    @Throws(ServiceException::class)
    fun updateListItem(listId: String, itemId: String, request: UpdateListItemRequest): AlexaListItem

    @Throws(ServiceException::class)
    fun deleteListItem(listId: String, itemId: String)
}
